### PR TITLE
[plugin.video.ted.talks] v5.0.0

### PR DESCRIPTION
**Pull request by moreginger**
### Description

- Support / require matrix
- Use m3u8 playlists
- Various bugfixes

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.ted.talks/addon.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    <description lang="en_GB">TED is a nonprofit devoted to spreading ideas, usually in the form of short, powerful talks (18 minutes or less). TED began in 1984 as a conference where Technology, Entertainment and Design converged, and today covers almost all topics — from science to business to global issues — in more than 100 languages. Meanwhile, independently run TEDx events help share ideas in communities around the world.</description>
+    <description lang="hu_HU">TED egy kicsi non-profit szervezet aminek célpontjában az Ötlet áll amit érdemes megosztani. A webhely kezdete 1984-re nyúlik vissza egy konferenciára ami 3 külön világot hozott össze: Technológia, Szórakozás, Tervezés. Azóta a kör csak bővült.</description>
+    <description lang="pt_PT">TED é uma organização sem fins lucrativos dedicada a 'Ideias com Valor'. Foi iniciada em 1984 como uma conferência que pretendia juntar pessoas de três mundos: Tecnologia, Entretenimento e Design. Desde ai tornou-se muito mais abrangente.</description>
+    <platform>all</platform>
+    <license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
```
Comment (enen92) [inline plugin.video.ted.talks/addon.xml]: GPL-2.0-only or GPL-2.0-or-later (please use a compliant spdx identifier for the license)

File: plugin.video.ted.talks/resources/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+# Dummy file to make this directory a package.
```
Comment (enen92) [inline plugin.video.ted.talks/resources/__init__.py]: not needed anymore with python3

File: plugin.video.ted.talks/resources/lib/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+# Dummy file to make this directory a package.
```
Comment (enen92) [inline plugin.video.ted.talks/resources/lib/__init__.py]: not needed anymore

File: plugin.video.ted.talks/resources/lib/model/__init__.py [hunk 4]
```diff
@@ -0,0 +1 @@
+# Dummy file to make this directory a package.
```
Comment (enen92) [inline plugin.video.ted.talks/resources/lib/model/__init__.py]: not needed

File: plugin.video.ted.talks/resources/lib/ted_talks.py [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+        plugin.report('%s = %s' % ('url', str(url)), level='debug')
+
+        subs_language = settings.get_subtitle_languages()
+        playlist, title, subs, info_labels = self.ted_talks.get_video_details(url=url, subs_language=subs_language)
+
+        playlist_file = os.path.join(xbmcvfs.translatePath("special://temp/"), 'ted_talks.m3u8')
+        with open(playlist_file, 'w', encoding='utf-8') as fh:
+            fh.write(playlist)
+
+        li = xbmcgui.ListItem(title, path=playlist_file)
```
Comment (enen92) [inline plugin.video.ted.talks/resources/lib/ted_talks.py]: consider to define `offscreen=True` in all your listitems. See https://codedocs.xyz/xbmc/xbmc/group__python__xbmcgui__listitem.html#ga15bcb930a4f91dd97f8ed7c633a20618

**Review by enen92 — CHANGES_REQUESTED** [review]
See comments

File: plugin.video.ted.talks/addon.xml (hunk 1 above)
Comment (moreginger) [reply plugin.video.ted.talks/addon.xml]: Thanks for the feedback. I've made the requested changes.

**Review by enen92 — APPROVED** [review]

**Pull request merged**